### DISCUSSION TITLE
People Search Results in Two Columns?

### DISCUSSION THREAD

**Discussion by LeeDrinkwater** [microsoft-search/pnp-modern-search#2063] [Q&A] [answered]
Hello,

I hope someone can help with what is probably a simple request. Is it possible to customise the People search results layout (making it custom) so that the results will display in two columns instead of only 1? 

Regardless of the component size setting of Extra Small to Extra Large, all people results will only display in one long vertical column. There is a large amount of whitespace around this column and so I think it would be useful to display in two or maybe 3 columns (if using small or extra small as component size). Can someone please advise if it is possible? I have the code exported from the People layout template but my coding skills are limited and I am not sure where/how to make the change to add a 2nd column.

Thanks for your help in advance

**Comment by FranckyC** [comment]
Hi @LeeDrinkwater, this is more a CSS/HTML question not directly related to the Web Part itself. Anyway, you can take a look at the "Cards" layout that does pretty much what you want. Just replace the `pnp-documentcard` component by the people card one and you should be fine.

**Reply by LeeDrinkwater** [reply]
Hello @FranckyC , thank you for your guidance. Apologies if this request should have gone elsewhere, or is it arguably not for this forum at all?

From your guidance, I am still a bit lost as to what to change as I am very much a noob with this subject so I appreciate your patience. In the code for the Cards layout, I can see **pnp-documentcard** declared several times so are you saying I should take a piece of this code and add it to the code from the People layout to replace something there to create the look I want? Perhaps replace the **pnp-persona** declaration?

`<pnp-documentcard class="template--card" 
                        data-ui-test-id="resultCard" 
                        data-item="{{JSONstringify item}}" 
                        data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.documentCardFields}}" 
                        data-enable-preview="{{@root.properties.layoutProperties.enablePreview}}" 
                        data-show-file-icon="{{@root.properties.layoutProperties.showFileIcon}}" 
                        data-is-compact="{{@root.properties.layoutProperties.isCompact}}" 
                        data-context="{{JSONstringify @root}}"
                        data-instance-id="{{@root.instanceId}}"
                        data-theme-variant="{{JSONstringify @root.theme}}"
                        data-allow-item-selection="{{@root.properties.itemSelectionProps.allowItemSelection}}"
                        data-item-key="{{@root.paging.currentPageNumber}}{{@index}}"                     
                        data-selected-keys="{{JSONstringify @root.selectedKeys}}"
                        data-index="{{@index}}"
                        >
                    </pnp-documentcard>`

`<mgt-person user-id="{{getUserEmail (slot item @root.slots.PersonQuery)}}" person-card="hover">
                                            <template>
                                                <pnp-persona 
                                                    data-image-url="\{{personImage}}" 
                                                    data-fields-configuration="{{JSONstringify @root.properties.layoutProperties.peopleFields}}" 
                                                    data-item="{{JSONstringify item}}" 
                                                    data-persona-size="{{@root.properties.layoutProperties.personaSize}}"
                                                    data-theme-variant="{{JSONstringify @root.theme}}"
                                                    data-instance-id="{{@root.instanceId}}"
                                                    data-context="{{JSONstringify @root}}">
                                                </pnp-persona>`

**Reply by wobba** [reply]
You don't need to use the documentcard. You can use any html/css you want to create the needed layout for your scenario. Look at the existing templates for inspiration on how to get started. The `<mgt-person>` control is documented at https://docs.microsoft.com/en-us/graph/toolkit/components/person

**Comment by paalolav** [comment, marked as answer]
@LeeDrinkwater Here's an example of what you could add to the People template in the style section to flex to three (or two based on how wide you make the card):

    .template--defaultList {
        flex-wrap: wrap;
        display: flex;
        justify-content: center;
        gap: 20px 25px;
    }

**Reply by LeeDrinkwater** [reply]
Thank you very much for providing this. It worked and is perfect. I appreciate the other responses were pointing me in a direction to figure it out for myself however my skills are very basic so I wasn't even sure where to start and had very limited time to try and solve it myself!

**Reply by adumith** [reply]
> @LeeDrinkwater Here's an example of what you could add to the People template in the style section to flex to three (or two based on how wide you make the card):
> 
> ```
> .template--defaultList {
>     flex-wrap: wrap;
>     display: flex;
>     justify-content: center;
>     gap: 20px 25px;
> }
> ```

Excuse where I should add the css?

I have the same issue.

**Reply by LeeDrinkwater** [reply]
@adumith , the additional code is added into the <style> section at the very top of the template. Mine looks like this with the recent change:

        mgt-person {
            --initials-background-color: {{@root.theme.palette.themePrimary}};
        }
        .template--defaultList {
           flex-wrap: wrap;
           display: flex;
           justify-content: center;
           gap: 20px 25px;
         }

**Reply by adumith** [reply]
> @adumith , the additional code is added into the <style> section at the very top of the template. Mine looks like this with the recent change:
> 
> ```
>     mgt-person {
>         --initials-background-color: {{@root.theme.palette.themePrimary}};
>     }
>     .template--defaultList {
>        flex-wrap: wrap;
>        display: flex;
>        justify-content: center;
>        gap: 20px 25px;
>      }
> ```

Hello!

Apply the new style and it works, however the columns are not populated neatly, see how they are sorted and how I expect them to be sorted.

How do I do it?
How is sorted...
![Picture1](https://user-images.githubusercontent.com/101891060/168642186-873193f6-e02a-4a9e-a978-a8aae618ec7e.png)

How I expect to sort 
![Picture2](https://user-images.githubusercontent.com/101891060/168642264-e467e195-cae4-49fb-b7d0-39594f50837e.png)

I have more question about this:

How avoid system/service accounts? I mean, I would like the result ONLY some real people.

Thanks again.

Sorry for being annoying :)

**Reply by markc75** [reply]
@adumith you might have already found your answer but ... set "justify-content: center;" to "justify-content: left;" and then inline style  on template--peopleListitem with style="width: 250px;"

**Reply by bergamini-ificonsulting** [reply]
Hi, I've done the described steps and everything works in the draft and published page. But then if I refresh the page it seems to lose some html in the webpart. I noticed for example that mgt-person is then sized 0x0. Inspecting the page code I saw that people data (pnp-persona) are not loaded and only the template is there. 
If I don't modify the style and just select the standard People layout in one column, everything is fine and I can see people even after reloading the page. My pnp-modern-search app version is 4.9.1.0.
Here the screenshot about a code compare between a just published page with people data (left) and a reloaded page without the people data (right). Do you think it could be a bug?

![image](https://github.com/microsoft-search/pnp-modern-search/assets/134713379/b32e816e-1810-46c2-8160-c43f711bb0b0)

Thanks for your help in advance.

**Comment by cluber22** [comment]
This is a nice answer, however when I view page 2 of the results the CSS resets. Is this to do with it pulling in fresh code and the CSS not been applied? Weirdly works on mobile.

**Comment by gretchunkim** [comment]
Use this :

```
<style>
    mgt-person {
        --initials-background-color: {{@root.theme.palette.themePrimary}};
    }
    .template--defaultList {
       flex-wrap: wrap;
       display: flex;
       justify-content: left;
       gap: 20px 25px;
    }
    .template--peopleListItem {
        width: 250px;
    }
</style>
```

**Comment by adumith** [comment]
Thank you for your reply.

This is my style. I wonder if there is a way to combine them with yours.

At this time I'm getting two columns and I need three.

Can you help me with that?

`
.template--defaultList{
    display: flex;
    flex-wrap: wrap;
}

.template--defaultList pnp-select{
    width: 380px;
    height: 110px;
    padding-bottom: 20px;
}

.ms-Persona-optionalText div {
    display: flex !important;
    font-size:large;
}

.ms-Persona-optionalText div a {
    padding-right: 10px;
    text-decoration: none;
}

`

Thank you so much again.

**Reply by kasperbolarsen** [reply]
please ask new questions in a new thread, this one has been answered
